pkg/controller/constants/image.go

```diff
@@ -14,20 +14,26 @@ const (
 	DefaultOperandImageOpenJ9 = "infinispan-for-openj9/server:latest"
 )
 
+var (
+	// Internal purpose variable for lazy initialization
+	defaultImage string
+)
+
 // GetDefaultInfinispanJavaImage returns default Infinispan Java image depends of the runtime architecture
 func GetDefaultInfinispanJavaImage() string {
+	// If the variable was already initialized, return it
+	if defaultImage != "" {
+		return defaultImage
+	}
+
 	// Full list of archs might be found here:
 	// https://github.com/golang/go/blob/release-branch.go1.10/src/go/build/syslist.go#L8
 	switch arch := runtime.GOARCH; arch {
 	case "ppc64le", "s390x":
-		return GetEnvWithDefault(OperandImageOpenJ9, DefaultOperandImageOpenJ9)
+		defaultImage = GetEnvWithDefault(OperandImageOpenJ9, DefaultOperandImageOpenJ9)
 	default:
-		// Try new name RELATED_IMAGE_OPENJDK first
-		retVal := GetEnvWithDefault(OperandImageOpenJDK, "")
-		if retVal != "" {
-			return retVal
-		}
-		// If RELATED_IMAGE_OPENJDK is not defined fallback on old name for backward comp
-		return GetEnvWithDefault("DEFAULT_IMAGE", DefaultOperandImageOpenJDK)
+		// Try new name RELATED_IMAGE_OPENJDK first, if not defined fallback on old name for backward comp
+		defaultImage = GetEnvWithDefault(OperandImageOpenJDK, GetEnvWithDefault("DEFAULT_IMAGE", DefaultOperandImageOpenJDK))
```

Comment:
@Crumby it's doesn't make sense for production deployment, but we could arrange DefaultOperandImageOpenJDK with the current Infinispan working image, but not latest, i.e set it to the "infinispan/server:12.0".

If we decide to make `DEFAULT_IMAGE` or `RELATED_IMAGE_OPENJDK` madatory, we can call `panic` if they are not present but `GetDefaultInfinispanJavaImage()` need to be validated  on operator startup, not lazy.

@rigazilla wdyt?
Otherwise looks good.

Reply:
> @Crumby it's doesn't make sense for production deployment, but we could arrange DefaultOperandImageOpenJDK with the current Infinispan working image, but not latest, i.e set it to the "infinispan/server:12.0".
> 

Yep, it should be 12 for master and 11 for 2.0.x

> If we decide to make `DEFAULT_IMAGE` or `RELATED_IMAGE_OPENJDK` madatory, we can call `panic` if they are not present but `GetDefaultInfinispanJavaImage()` need to be validated on operator startup, not lazy.
> 
> @rigazilla wdyt?
> Otherwise looks good.

@Crumby seems you saved the _mandatory_ solution for another issue, right?
lgtm for fixing the initial issue, though removing all the DefaultOperand* in the code and fail fast should be a safer solution imo

Reply:
It looks like we walked full circle at this point. So my suggestion is:

1) Drop the image fields and change the default from `latest` to `12` in this PR. This will cause that tests executed with Data Grid Operator will not default to Infinispan but to `DEFAULT_IMAGE` of the Operator. Tests running the operator locally will still default to Infinispan 12. Should be non-invasive.
2) Create an issue and another PR for `GetDefaultInfinispanJavaImage()` to panic if `DEFAULT_IMAGE` isn't defined as safeguard preventing regression. This change isn't trivial because it'll mean that in order to run the tests from IDE the variable `DEFAULT_IMAGE` will have to be defined before the tests are executed.

Does it make sense?